src/WebJobs.Script.WebHost/WebHostServiceCollectionExtensions.cs

```diff
@@ -179,6 +179,9 @@ public static void AddWebJobsScriptHost(this IServiceCollection services, IConfi
 
             services.TryAddSingleton<IDependencyValidator, DependencyValidator>();
             services.TryAddSingleton<IJobHostMiddlewarePipeline>(s => DefaultMiddlewarePipeline.Empty);
+
+            // Add AzureStorageProvider to WebHost (also needed for ScriptHost)
+            services.AddAzureStorageProvider();
```

Comment:
Let's review this service registration to make sure the behavior is as intended. With this work, all blob operations go through IAzureStorageProvider, which internally uses the IConfiguration. We have two scopes (Root WebHost and ScriptHost) at play here and the used IConfiguration object in IAzureStorageProvider depends on where we register this service. 
ScriptHost adds more Configuration builders (i.e. HostJson) - will IAzureStorageProvider need a dependency on these extra configurations? Right now, it's enough to register IAzureStorageProvider at WebHost layer since Environment Variables are included.